sale_exception/data/sale_exception_data.xml

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<!-- License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl). -->
+<odoo noupdate="1">
+    <record forcecreate="True" id="ir_cron_test_orders" model="ir.cron">
+        <field name="name">Test Draft Orders</field>
+        <field name="active" eval="False" />
+        <field name="user_id" ref="base.user_root" />
+        <field name="interval_number">20</field>
+        <field name="interval_type">minutes</field>
+        <field name="numbercall">-1</field>
+        <field eval="False" name="doall" />
+        <field name="model_id" ref="sale.model_sale_order" />
+        <field name="state">code</field>
+        <field name="code">model.test_all_draft_orders()</field>
+    </record>
+    <record id="excep_no_zip" model="exception.rule">
+        <field name="name">No ZIP code on destination</field>
+        <field name="description">No ZIP code on destination</field>
+        <field name="sequence">50</field>
+        <field name="model">sale.order</field>
+        <field name="code">if not obj.partner_shipping_id.zip: failed=True</field>
+        <field name="active" eval="False" />
+    </record>
+    <record id="excep_no_stock" model="exception.rule">
+        <field name="name">Not Enough Virtual Stock</field>
+        <field name="description">Not Enough Virtual Stock</field>
+        <field name="sequence">50</field>
+        <field name="model">sale.order.line</field>
+        <field
+            name="code"
+        >if obj.product_id and sale_line.product_id.type == 'product' and obj.product_id.with_context(warehouse=obj.order_id.warehouse_id.id).virtual_available &lt; obj.product_uom_qty: failed=True</field>
```

Comment:
```suggestion
        >if obj.product_id and obj.product_id.type == 'product' and obj.product_id.with_context(warehouse=obj.order_id.warehouse_id.id).virtual_available &lt; obj.product_uom_qty: failed=True</field>
```

Reply:
The check must be updated in 14.0 to use  the field virtual_quantity_at_date of the sale line to do the check. 

See : https://github.com/odoo/odoo/blob/1604f23afdb5c18d00147e25d0bc40cd0b57620c/addons/sale_stock/models/sale_order.py#L270